src/hdskins/java/com/voxelmodpack/hdskins/mixin/MixinGuiMainMenu.java

```diff
@@ -15,19 +16,17 @@
 @Mixin(GuiMainMenu.class)
 public class MixinGuiMainMenu extends GuiScreen {
 
-    private static final int SKINS = 5000;
-
     @Inject(method = "initGui()V", at = @At("RETURN"))
     private void onInit(CallbackInfo ci) {
-        ItemStack itemStack = new ItemStack(Items.LEATHER_LEGGINGS);
-        Items.LEATHER_LEGGINGS.setColor(itemStack, 0x3c5dcb);
-        this.buttonList.add(new GuiItemStackButton(SKINS, width - 50, height - 50, itemStack));
+        addButton(new IconicButton(width - 50, height - 50, sender -> {
+            mc.displayGuiScreen(HDSkinManager.INSTANCE.createSkinsGui());
+        }).setIcon(new ItemStack(Items.LEATHER_LEGGINGS), 0x3c5dcb));
     }
 
     @Inject(method = "actionPerformed(Lnet/minecraft/client/gui/GuiButton;)V", at = @At("RETURN"))
     private void onActionPerformed(GuiButton button, CallbackInfo ci) {
-        if (button.id == SKINS) {
```

Comment:
The whole purpose of IGuiAction is to not have to set a unique Id. Previously we couldn't know if another mod was adding a button with the same id as us in which case you would have the same option.

Right now it doesn't interfere with any other actions because the mixin injects at the method head. We can add a called to info.cancel() though, after handling the button be sure.

Reply:
Other mods still use the old way, especially other mods that completely [rewrite the main menu](https://minecraft.curseforge.com/projects/custom-main-menu). I know it is impossible to not have duplicate button ids, but setting the ID to 0 won't help anyone.

Reply:
You want me to set it to -1 instead? There is no id you can pick that would not result in a clash _eventually_.

Reply:
Just use the one I had earlier. I think it was `5000`